### PR TITLE
Bridge pot currency swap implementation

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

File: crates/bridge-pot-currency-swap/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/// One to one bridge pot currency swap logic.
+pub struct OneToOne<AccountIdFrom, AccountIdTo, ConfigT, CurrencyFrom, CurrencyTo>(
+    PhantomData<(
+        AccountIdFrom,
+        AccountIdTo,
+        ConfigT,
+        CurrencyFrom,
+        CurrencyTo,
+    )>,
+)
```
Comment (MOZGIII) [inline crates/bridge-pot-currency-swap/src/lib.rs:31]: I think we could move all those params to `Config` for better UX. We already have it, so why don't just move the rest of the params (`AccountIdFrom`, `AccountIdTo`, `CurrencyFrom`, `CurrencyTo`) there.

File: crates/bridge-pot-currency-swap/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        let amount = imbalance.peek();
+
+        CurrencyFrom::resolve_creating(&ConfigT::PotFrom::get(), imbalance);
+
+        let imbalance = CurrencyTo::withdraw(
+            &ConfigT::PotTo::get(),
+            amount.into(),
+            WithdrawReasons::TRANSFER,
+            ExistenceRequirement::AllowDeath,
+        )?;
```
Comment (MOZGIII) [inline crates/bridge-pot-currency-swap/src/lib.rs:62]: We might want to disallow the creation (re `resolve_creating`) and removal (re `ExistenceRequirement::AllowDeath`) of the bridge account.

**Review by MOZGIII — APPROVED** [review]
I have applied the changes I wanted, so now it looks good to me. @dmitrylavrenov please provide your review and we can merge if all is good.

File: crates/bridge-pot-currency-swap/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    /// The account to land the balances to when receiving the funds as part of the swap operation.
+    type PotFrom: Get<Self::AccountIdFrom>;
+
+    /// The account to take the balances from when sending the funds as part of the swap operation.
+    type PotTo: Get<Self::AccountIdTo>;
+}
+
+/// A [`primitives_currency_swap::CurrencySwap`] implementation that does the swap using two
+/// "pot" accounts for each of end swaped currencies.
```
Comment (dmitrylavrenov) [inline crates/bridge-pot-currency-swap/src/lib.rs:47]:
```suggestion
/// "pot" accounts for each of end swapped currencies.
```

File: crates/bridge-pot-currency-swap/src/existence_required.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+/// A marker type for the implementation that requires pot accounts existence.
+pub enum Marker {}
+
+/// An error that can occur while doing the swap operation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Error<ImbalanceFrom> {
+    /// Unable to resolve the incoming balance into the corresponding pot.
+    ResolvingIncomingImbalance(ImbalanceFrom),
+    /// Unable to withdraw the outpoing balance from the corresponding pot.
```
Comment (dmitrylavrenov) [inline crates/bridge-pot-currency-swap/src/existence_required.rs:18]:
```suggestion
    /// Unable to withdraw the outgoing balance from the corresponding pot.
```

**Comment by dmitrylavrenov** [comment]
Awesome proposed changes, let's merge!

**Pull request merged**